### PR TITLE
Autocomplete variable parameter with the variable selected in the variable editor

### PR DESCRIPTION
**Pull request by D8H**
Also:
- Create new variables with the type "Number" by default instead of "String".
- Use the type of the previous child to add new variables to arrays.

### CODE REVIEW

File: newIDE/app/src/VariablesList/VariablesEditorDialog.js [hunk 1]
```diff
@@ -11,12 +11,26 @@ import useDismissableTutorialMessage from '../Hints/useDismissableTutorialMessag
...
+  nodeId: string,
+  variablesContainer: gdVariablesContainer
+): string => {
+  const variableContext = getVariableContextFromNodeId(
+    nodeId,
+    variablesContainer
+  );
+  const variablePath = variableContext.lineage.map(variable => variable.name);
+  variablePath.push(variableContext.name);
+  return variablePath.join('.');
```
Comment (4ian) [inline newIDE/app/src/VariablesList/VariablesEditorDialog.js:28]: Will this work with arrays?

File: newIDE/app/src/VariablesList/VariablesList.js [hunk 2]
```diff
@@ -523,12 +524,23 @@ const VariablesList = (props: Props) => {
...
+  const [selectedNodes, doSetSelectedNodes] = React.useState<Array<string>>([]);
+  const setSelectedNodes = React.useCallback(
+    (nodes: Array<string> | ((nodes: Array<string>) => Array<string>)) => {
+      nodes = Array.isArray(nodes) ? nodes : nodes(selectedNodes);
+      doSetSelectedNodes(nodes);
+      if (onSelectedVariableChange) {
+        onSelectedVariableChange(nodes);
+      }
+    },
+    [onSelectedVariableChange, selectedNodes]
```
Comment (4ian) [inline newIDE/app/src/VariablesList/VariablesList.js:541]: Do you really need to depend on `selectedNodes`? This means any state change will also change all the callbacks using `setSelectedNodes`. Previously, this was a stable function.

Comment (4ian) [reply newIDE/app/src/VariablesList/VariablesList.js:541]: Try this instead:

```
const [selectedNodes, doSetSelectedNodes] = React.useState<Array<string>>([]);
const setSelectedNodes = React.useCallback(
  (nodes: Array<string> | ((nodes: Array<string>) => Array<string>)) => {
    // Use the functional update form to access the current state directly
    doSetSelectedNodes(currentSelectedNodes => {
      const newNodes = Array.isArray(nodes) ? nodes : nodes(currentSelectedNodes);
      if (onSelectedVariableChange) {
        onSelectedVariableChange(newNodes);
      }
      return newNodes;
    });
  },
  [onSelectedVariableChange] // Remove selectedNodes from dependencies
);
```

File: newIDE/app/src/VariablesList/VariablesEditorDialog.js (hunk 1 above)
Comment (D8H) [reply newIDE/app/src/VariablesList/VariablesEditorDialog.js:28]: Yes, for array children, it does something like `MyArray.3`. It's probably not what users want if they added an array and have one child selected, but I guess that's ok.

Do you think we should remove the child for these cases?
I guess, at some point, the field could choose according to the type of variable that is needed.

**Comment by arthuro555** [comment]
> Create new variables with the type "Number" by default instead of "String".

Hmmm... Is this decision based on user research? 

Personally I find myself using strings more often than numbers, and to prefer strings as a default, since I can still write a number in there, then change the type. I personally do not find myself always going through the fields I configure from left to right/top to bottom, but to jump ahead then back depending to fill in fields I assume I might forget first, and the value I settled on setting the variable to is something I am more likely to forget than they type I want to give the variable.

In fact I never was frustrated by the default type being a string, but am very often irritated by Arrays children not being strings by default....

**Comment by D8H** [comment]
> Hmmm... Is this decision based on user research?
>
> Personally I find myself using strings more often than numbers, and to prefer strings as a default, since I can still write a number in there, then change the type.

Looking at examples, games mostly use number variables. Strings are usually directly set on text objects. I guess that strings variable are mostly used for the state of FSM.

Could it be that you often work on network related features which probably requires things like string id?

> In fact I never was frustrated by the default type being a string, but am very often irritated by Arrays children not being strings by default....

Good new for you, arrays now use the previous child type.

**Comment by D8H** [comment]
These changes has already be merged by:
- https://github.com/4ian/GDevelop/pull/6459

**Pull request closed without merging**